kserve-module/tests/e2e/conftest.py

```
@@ -49,6 +51,77 @@ class ClusterInfo:
+# ---------------------------------------------------------------------------
+# Debug helpers (temporary — drop this commit after CI investigation)
+# ---------------------------------------------------------------------------
+def _dbg(msg):
+    """Print a debug message to stderr so pytest captures it."""
+    print(f"[DEBUG] {msg}", file=sys.stderr, flush=True)
+
+
+def _dbg_cr_state(kubectl_bin, name=KSERVE_CR_NAME, label=""):
+    """Dump Kserve CR state for debugging."""
+    prefix = f"[DEBUG][{label}]" if label else "[DEBUG]"
+    result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)